### PR TITLE
Mes 2949 submit incomplete rsis entry

### PR DESCRIPTION
**Pull request by edwardwilson**
# Description and relevant Jira numbers
Changes to the logic to detect if the test is auto save or not
## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+indent_size = 2
+
+# We recommend you to keep these unchanged
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (rossfauldsbjss) [inline .editorconfig:17]: which trim_trailing_whitespace wins?

File: src/functions/uploadToRSIS/application/__tests__/result-client.spec.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            checkDigit: 2,
+          },
+        },
+        testSummary: {
+          D255: true,
+        },
+      };
+
+      // ASSERT
+      expect(isRecordAutosaved(mockData as StandardCarTestCATBSchema)).toBeTruthy();
```
Comment (MarkPatt89) [inline src/functions/uploadToRSIS/application/__tests__/result-client.spec.ts]: Can we change the assertion to not use `toBeTruthy()`? Opt for `toBe(true)` instead.

Here's a post I saved ages ago if you want to further understand why. https://vincenttunru.com/toBeTruthy-vs-toBe-true/

File: src/functions/uploadToRSIS/application/__tests__/result-client.spec.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+              checkDigit: 2,
+            },
+          },
+          testSummary: {
+            D255: true,
+          },
+        };
+
+        // ASSERT
+        expect(isRecordAutosaved(mockData as StandardCarTestCATBSchema)).toBeTruthy();
```
Comment (MarkPatt89) [inline src/functions/uploadToRSIS/application/__tests__/result-client.spec.ts]: Change it here as well and in all other relevant test assertions.

**Review by rossfauldsbjss — APPROVED** [review]
looks good

**Review by MarkPatt89 — CHANGES_REQUESTED** [review]
It looks good, just change out those assertions.

File: .editorconfig (hunk 1 above)
Comment (edwardwilson) [reply .editorconfig:17]: This the editor config out go the app code

Comment (edwardwilson) [reply .editorconfig:17]: Also I the trim whitespace will be true for all files except for md files.

File: src/functions/uploadToRSIS/application/result-client.ts [hunk 4]
```diff
@@ -163,15 +164,21 @@ const mapHTTPErrorToDomainError = (err: AxiosError): Error => {
...
 };
 
-const isRecordAutosaved = (test: StandardCarTestCATBSchema): BooleanAsNumber => {
-  // If the following properties are provided, we safely assume this is not an autosaved record.
-  // As testSummary maybe undefined, we use the non-null assertion operator.
+export const isRecordAutosaved = (test: StandardCarTestCATBSchema): BooleanAsNumber => {
+  const testDate = moment(new Date(test.journalData.testSlotAttributes.start));
+
+  // Two weeks or older is always flagged as autosave
+  if (moment().diff(testDate, 'days') >= 14) {
```
Comment (rstepnowski) [inline src/functions/uploadToRSIS/application/result-client.ts:171]: Did we not agree that the >=14 days condition should be a part of the conditions below? Or did I recall it incorrectly?

File: src/functions/uploadToRSIS/application/result-client.ts [hunk 5]
```diff
@@ -163,15 +164,21 @@ const mapHTTPErrorToDomainError = (err: AxiosError): Error => {
...
+  // We assume that when these properties are null then the test is autosaved
   return (
-    get(test, 'testSummary.additionalInformation', false) &&
-    get(test, 'testSummary.candidateDescription', false) &&
-    get(test, 'testSummary.identification', false) &&
-    get(test, 'testSummary.independentDriving', false) &&
-    get(test, 'testSummary.routeNumber', false) &&
-    get(test, 'testSummary.weatherConditions', false))
-     ? 0 : 1;
+    (get(test, 'testSummary.additionalInformation', false) &&
```
Comment (rstepnowski) [inline src/functions/uploadToRSIS/application/result-client.ts]: Is this not an extra parenthesis (at the beginning of this line) - there is on return level ?

File: src/functions/uploadToRSIS/application/result-client.ts (hunk 4 above)
Comment (edwardwilson) [reply src/functions/uploadToRSIS/application/result-client.ts:171]: We said if anything was older than 14 days we would flag as no write up regardless as it has clearly not been completed.

If for whatever reason we get the test result within the two week window we check the properties if are empty to decide if to set no write up

**Review by MarkPatt89 — APPROVED** [review]

File: src/functions/uploadToRSIS/application/result-client.ts (hunk 4 above)
Comment (rstepnowski) [reply src/functions/uploadToRSIS/application/result-client.ts:171]: Ah, ok - that's fine, just wanted to confirm 👍

**Review by rstepnowski — APPROVED** [review]

**Comment by TomBillingtonUK** [comment]
Just a note on this one - It shouldn't have been merged with approval from both teams, there are only team j approvals on the pr

**Pull request merged**